### PR TITLE
Migrate to another package of apache http client library

### PR DESCRIPTION
**Pull request by rpdome**
Related PR
android-complete: https://github.com/AzureAD/android-complete/pull/98/files
MSAL: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1436/files
ADAL: https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1613
Broker: https://github.com/AzureAD/ad-accounts-for-android/pull/1590

---

Apache HttpClient cannot be used directly as a dependency in android, because Android already use (part of) it. 

Therefore, we're moving to another Apache Httpclient package whose package name is changed (as suggested [here](https://hc.apache.org/httpcomponents-client-4.5.x/android.htm))

More info here: https://hc.apache.org/httpcomponents-client-4.5.x/android.html
The repo itself: https://github.com/smarek/httpclient-android

### CODE REVIEW

File: gradle/versions.gradle [hunk 1]
```diff
@@ -42,7 +42,7 @@ ext {
     robolectricVersion = "4.5.1"
     uiAutomatorVersion = "2.2.0"
     msal4jVersion = "1.10.0"
-    apacheHttpVersion = "4.5.13"
```
Comment (AdamBJohnsonx) [inline gradle/versions.gradle:45]: msebraApacheHttpClientVersion?  You probably shouldn't arbritrarily switch these...

Comment (AdamBJohnsonx) [reply gradle/versions.gradle:45]: Also, this change is causing build failures in your consumers.  Are there updates for them?

Comment (rpdome) [reply gradle/versions.gradle:45]: yes - for MSAL and broker, I'm going to bundle them with the other PR.
For ADAL - I 'll open a separate one.

File: gradle/versions.gradle [hunk 2]
```diff
@@ -42,7 +42,7 @@ ext {
...
+    msebraApacheHttpClientVersion = "4.5.8"
```
Comment (shahzaibj) [inline gradle/versions.gradle:45]:
```suggestion
    mseberaApacheHttpClientVersion = "4.5.8"
```

File: common/build.gradle [hunk 3]
```diff
@@ -172,7 +173,7 @@ dependencies {
     testImplementation "androidx.test:core:$rootProject.ext.androidxTestCoreVersion"
 
     // For powermockito!
-    testImplementation "org.apache.httpcomponents:httpclient:$rootProject.ext.apacheHttpVersion"
+    testImplementation "cz.msebera.android:httpclient:$rootProject.ext.msebraApacheHttpClientVersion"
```
Comment (shahzaibj) [inline common/build.gradle:176]: Does the implementation not carry over to tests as well?

**Review by shahzaibj — APPROVED** [review]
left minor comments

**Review by AdamBJohnsonx — APPROVED** [review]

**Pull request merged**